UniWatch/DataAccess/LectureManager.cs

```diff
@@ -69,19 +78,41 @@ public IEnumerable<StudentAttendance> GetStudentReport(int classId, int studentI
         }
 
         /// <summary>
-        /// Updates and saves the given lecture
+        /// Updates/Overrides the values in a lecture
         /// </summary>
-        /// <param name="lecture">The lecture to update</param>
+        /// <param name="lectureId">The id of the lecture to update</param>
+        /// <param name="updates">The list of updates to make</param>
         /// <returns>The updated lecture</returns>
-        public Lecture Update(Lecture lecture)
+        public Lecture Update(int lectureId, IEnumerable<UpdateLectureItem> updates)
         {
-            var existing = _db.Lectures.Find(lecture.Id);
+            var lecture = _db.Lectures.Find(lectureId);
 
             // Doesn't exist
-            if(existing == null)
-                throw new InvalidOperationException("Error updating lecture");
+            if(lecture == null)
+                throw new InvalidOperationException("Error updating lecture. Id not found.");
+
+            var attendanceMap = new Dictionary<int, StudentAttendance>(lecture.Attendance.Count);
+            foreach(var attendance in lecture.Attendance)
+                attendanceMap.Add(attendance.Student.Id, attendance);
+
+            foreach (var item in updates)
+            {
+                StudentAttendance attendance;
+                if(attendanceMap.TryGetValue(item.StudentId, out attendance))
+                {
+                    if(attendance.Present != item.Present)
+                    {
+                        // Only update if changed
+                        attendance.Present = item.Present;
+                        _db.Attendance.Attach(attendance);
+                        _db.Entry(attendance).Property(a => a.Present).IsModified = true;
```

Comment:
Only updates the `present` property of the lecture attendance. The method should probably be split into two separate methods (update and override).